### PR TITLE
Task/jenkins 39478 dropdown

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Description**
- [x] This branch is built off the one waiting to be merged in #111 - will remerge once that is ready.
- Wanted to get some feedback on this one before I go too much further. Has a bunch of console stuff that needs to be cleaned up before merge but am leaving it in for now for debugging purposes.
- See [JENKINS-39478](https://issues.jenkins-ci.org/browse/JENKINS-39478).

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
cc @sophistifunk curious to get your feedback on this one, and perhaps for you to play with it a bit to see it feels natural to you. Tried to emulate `select` behavior pretty closely.

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "url": "https://github.com/jenkinsci/jenkins-design-language.git"
   },
   "dependencies": {
+    "lodash.debounce": "4.0.8",
     "moment": "2.15.1",
     "moment-duration-format": "1.3.0",
     "react": "15.3.2",
```
Comment (cliffmeyers) [inline package.json]: Writing my own `debounce` seemed silly but I can see maybe just pulling this directly into our repo to cut down on the dependency headaches.

File: src/js/KeyCodes.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    SPACEBAR: 32,
+    PAGE_UP: 33,
+    PAGE_DOWN: 34,
+    END: 35,
+    HOME: 36,
+    ARROW_LEFT: 37,
+    ARROW_UP: 38,
+    ARROW_RIGHT: 39,
+    ARROW_DOWN: 40,
+};
```
Comment (sophistifunk) [inline src/js/KeyCodes.js]: We should probably double-check how page up/down and home/end work across keyboard layouts and OSs to make sure it behaves as we expect. I've been reading a lot recently about others' problems with it.

**Comment by sophistifunk** [comment]
screen shot 2016-11-10 at 9 47 57 am Should this broken image be a little arrow or something? Looks like it's pointing to a missing `/foo.png`.

**Comment by sophistifunk** [comment]
Looks good once I got the stupid thing to install, thanks node-sass. I'm glad *that* problem is fixed in master.

bbm

**Comment by cliffmeyers** [comment]
@sophistifunk fixed the broken image in bf1781b. Am restyling this per the buttons we currently have, although that stuff is going to get reskinned once style guide is finalized.

**Pull request merged**